src/cgt/src/ClockGating.cpp

```diff
@@ -0,0 +1,1427 @@
+    if (correct_conds.empty()) {
+      continue;
+    }
+
+    logger_->info(
+        CGT,
+        4,
+        "Accepted clock {} '{}' for instance {}",
+        clk_enable ? "enable" : "disable",
+        combinedGatingCondNames(
+            network, correct_conds.begin(), correct_conds.end(), clk_enable),
+        network->name(instance));
+
+    {
+      DebugScopedTimer timer(exist_check_time_,
+                             logger_,
+                             CGT,
+                             "clock_gating",
+                             3,
+                             "Check if gate condition already exists time: {}");
+      for (auto net : correct_conds) {
+        for (auto idx : net_to_accepted[net]) {
+          auto& [conds, insts, en] = accepted_gates[idx];
+          if (clk_enable == en
+              && std::includes(correct_conds.begin(),
+                               correct_conds.end(),
+                               conds.begin(),
+                               conds.end())) {
+            logger_->info(CGT,
+                          5,
+                          "Extending previously accepted clock {} '{}' to '{}'",
+                          clk_enable ? "enable" : "disable",
+                          combinedGatingCondNames(
+                              network, conds.begin(), conds.end(), clk_enable),
+                          combinedGatingCondNames(network,
+                                                  correct_conds.begin(),
+                                                  correct_conds.end(),
+                                                  clk_enable));
+            conds = correct_conds;
+            for (auto net : correct_conds) {
+              auto& gates = net_to_accepted[net];
+              auto it = std::lower_bound(gates.begin(), gates.end(), idx);
+              if (it == gates.end() || *it != idx) {
+                gates.insert(it, idx);
+              }
+            }
+            insts.insert(insts.end(), instance);
+            inserted = true;
+            break;
+          }
+        }
+        if (inserted) {
+          break;
+        }
+      }
+    }
+    if (!inserted) {
+      accepted_gates.emplace_back(
+          correct_conds, std::vector<sta::Instance*>{instance}, clk_enable);
+      for (auto net : correct_conds) {
+        net_to_accepted[net].push_back(accepted_gates.size() - 1);
+      }
+    }
+  }
+
+  for (auto& [correct_conds, instances, clk_enable] : accepted_gates) {
+    if (instances.size() >= min_instances_) {
+      insertClockGate(instances, correct_conds, clk_enable);
+    } else {
+      debugPrint(
+          logger_,
+          CGT,
+          "clock_gating",
+          1,
+          "Not enough instances ({}) for clock gate: {}",
+          instances.size(),
+          combinedInstanceNames(network, instances.begin(), instances.end()));
+    }
+  }
+
+  dump("post");
+
+  debugPrint(logger_, CGT, "clock_gating", 1, "Clock gating completed.");
```

Comment:
Just wrap this in a single logger->debugCheck rather than repeatedly checking the same condition.  It will also be more compact.  logger->debug can be used inside.